src/util/internal/asn1.h

```diff
@@ -69,8 +69,8 @@ enum { name_attr_tandv_length = 1 + 1 + name_oid_length + name_string_length };
 enum { rdn_length = 1 + 1 + name_attr_tandv_length };                                   // tag(1) + length(1) + name_attr_tandv_length
 enum { name_length = 1 + 1 + rdn_length };                                              // tag(1) + length(1) + rdn_length
 
-enum { utctime_length = 1 + 1 + UTC_LENGTH };                                           // tag(1) + length(1) + content(13) 'YYMMDDhhssmmZ'
-enum { validity_length = 1 + 1 + 2*utctime_length };                                    // tag(1) + length(1) + 2*utctime_length (one for notBefore, other for notAfter)
+enum { generalizedtime_length = 1 + 1 + GENERALIZEDTIME_LENGTH };                                           // tag(1) + length(1) + content(13) 'YYYYMMDDhhmmssZ'
```

Comment:
Should content be changed to 15 here? And the only other thing that we may want to change is in `test/server_setup.h` giving the serialized certificate the default reserved value, although it doesn't really matter. This is awesome, thanks!

Reply:
Yes, good catch on that comment. That should be `content(15)`.


And another good call about the `test/server_setup.h`. I'll just leave the bytes as they are, but label them as "reserved", rather than "server ID" and "expiry".

I'll force push this branch making these changes now.

Thanks!